pyproject.toml

```diff
@@ -1,7 +1,92 @@
 [build-system]
-requires = ["setuptools", "wheel"]
+requires = ["setuptools>=67.0.0"]
 build-backend = "setuptools.build_meta"
 
+
+[project]
+name = "pipenv"
+authors = [
+  {name = "Pipenv maintainer team", email = "distutils-sig@python.org"},
+]
+description = "Python Development Workflow for Humans."
+readme = "README.md"
+requires-python = ">=3.7"
+license = {file = "LICENSE"}
+classifiers=[
+  "License :: OSI Approved :: MIT License",
+  "Programming Language :: Python",
+  "Programming Language :: Python :: 3",
+  "Programming Language :: Python :: 3.7",
+  "Programming Language :: Python :: 3.8",
+  "Programming Language :: Python :: 3.9",
+  "Programming Language :: Python :: 3.10",
+  "Programming Language :: Python :: 3.11",
+  "Programming Language :: Python :: Implementation :: CPython",
+  "Programming Language :: Python :: Implementation :: PyPy",
+]
+dependencies = [
+  "certifi",
+  "setuptools>=67.0.0",
+  "virtualenv>=20.24.2",
+]
+dynamic = ["version"]
+
+
+[project.optional-dependencies]
+dev = [
+  "towncrier",
+  "beautifulsoup4",
+  "sphinx",
+  "flake8>=3.3.0,<4.0",
+  "black==23.3.0",
+  "parver",
+  "invoke",
+]
+tests = [
+  "pytest>=5.0",
+  "pytest-timeout",
+  "pytest-xdist",
+  "flaky",
+  "mock",
+]
+
+
+[project.scripts]
+pipenv = "pipenv:cli"
+pipenv-resolver = "pipenv.resolver:main"
+
+
+[project.urls]
+Homepage = "https://github.com/pypa/pipenv"
+Documentation = "https://pipenv.pypa.io/en/latest/"
+Source = "https://github.com/pypa/pipenv.git"
+
+
+[tool.setuptools.packages.find]
+where = ["."]
+exclude = ["tests*", "tests.*", "tasks*", "tasks.*"]
+
+
+[tool.setuptools.package-data]
+"*" = ["LICENSE", "NOTICES"]
+"pipenv.patched.safety" = ["VERSION", "safety-policy-template.yml"]
+"pipenv.patched.pip._vendor.certifi" = ["*.pem"]
+"pipenv.patched.pip._vendor.requests" = ["*.pem"]
+"pipenv.patched.pip._vendor.distlib" = [
+  "t32.exe",
+  "t64.exe",
+  "w32.exe",
+  "w64.exe",
+]
+"pipenv.vendor.ruamel" = ["yaml"]
+
+
+[tool.setuptools.dynamic]
+version = {attr = "pipenv.__version__"}
+
```

Comment:
Please add __version_ to  gitgnore.

Reply:
Sure. I mean `__version__.py` is a file that belongs to the repo, but I can see how you might not want someone to accidentally overwrite and commit it.

Reply:
@oz123 I just want to make sure this isn't going to break
```python
if commit:
    ctx.run(f"git add {version_file.as_posix()}")
    log("Committing...")
    ctx.run('git commit -s -m "Bumped version."')
```
in `release.py`.

I'm not certain if ignored files have to be force added the first time or every time after.

Reply:
Perhaps I am confused about `pipenv.__version__` vs. `pipenv.__version__.__version__`...

The `pip` project stores `pip.__version__` in a `pip.__init__.py` file while this project stores the version in a separate `pipenv.__version__.py` file.
* https://github.com/pypa/pip/blob/main/src/pip/__init__.py#L3 vs.
* https://github.com/pypa/pipenv/blob/main/pipenv/__version__.py#L5

Will the following lines create the variable `pipenv.__version__` or `pipenv.__version__.__version__`?
```python
dynamic = ["version"]
# and
version = {attr = "pipenv.__version__"}
```
Was this pull request created with a tool like https://pypi.org/project/pyproject-migrator or was it done manually?

Reply:
It was done manually. `__version__.py` files are pretty standard for a number of projects. The import of `__version__` from `__version__.py` is already a part of the package. What the lines referenced do is read the version from the file for the purpose of the build. If you review the build/test artifacts, you can see this is doing exactly what is expected.

Reply:
@oz123 Do you know if the script will still work properly if the file is added to `.gitignore`?

Reply:
I don't know if it will break. Need to test it, of course. I asked to remove it, because I thought that file is auto generation by setuptools_scm if your configuration.

Reply:
Latest version doesn't use `setuptools_scm`. Thought I'd use it, but when investigated the current release process, I realized it would require much more work than what I intended as the scope of this issue.

Reply:
@oz123 We can consider later to move to `setuptools_scm`, but it would require changing the entire automated-release process.

Right now, `__version__.py` is used to tell both the package (via `__init__.py`) and the build what the current version is. It's a file the automated-release process doesn't create but rather updates each time.

Reply:
Adding to `.gitignore` could be innocuous if `git add` without the `--force` option will update the existing file. Otherwise, one would also have to change the automated-release file with this option. Since this package actually includes `__version__.py` in its repo, my opinion would be to leave as-is until such a time that one changes the release process.

Reply:
@oz123 Your review is still pending.